bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx

```diff
@@ -206,271 +206,299 @@ export default function Whiteboard(props) {
-                persistShape(childShape, whiteboardId)
-              });
-            });
-            const pageShapes = e.state.document.pages[e.getPage()?.id]?.shapes;
-            let shapesIdsToRemove = findRemoved(Object.keys(shapes), Object.keys(pageShapes))
-            if (shapesIdsToRemove.length) {
-              // add a little delay, wee need to make sure children are updated first
-              setTimeout(() => removeShapes(shapesIdsToRemove, whiteboardId), 200);
-            }
-          }}
+    if (curPageId) {
+      app.changePage(curPageId);
+      if (slidePosition.zoom === 0) {
+        // first load, center the view to fit slide
+        const cameraFitSlide = calculateCameraFitSlide();
+        app.setCamera(cameraFitSlide.point, cameraFitSlide.zoom);
+        setCameraFitSlide(cameraFitSlide);
+      } else {
+        app.setCamera([slidePosition.xCamera, slidePosition.yCamera], slidePosition.zoom);
+        setIsZoomed(true);
+      }
+    }
+  };
 
-          onChangePage={(app, s, b, a) => {
-            if (app.getPage()?.id !== curPageId) {
-              skipToSlide(Number.parseInt(app.getPage()?.id), podId)
-            }
-          }}
-          onCommand={(e, s, g) => {
-
-            if (s.includes('move_to_page')) {
-              let groupShapes = [];
-              let nonGroupShapes = [];
-              let movedShapes = {};
-              e.selectedIds.forEach(id => {
-                const shape = e.getShape(id);
-                if (shape.type === 'group')
-                  groupShapes.push(id);
-                else
-                  nonGroupShapes.push(id);
-                movedShapes[id] = e.getShape(id);
-              });
-              //remove shapes on origin page
-              let idsToRemove = nonGroupShapes.concat(groupShapes);
-              removeShapes(idsToRemove, whiteboardId);
-              //persist shapes for destination page
-              const newWhiteboardId = curPres.pages.find(page => page.num === Number.parseInt(e.getPage()?.id)).id;
-              let idsToInsert = groupShapes.concat(nonGroupShapes);
-              idsToInsert.forEach(id => {
-                persistShape(movedShapes[id], newWhiteboardId);
-                const children = movedShapes[id].children;
-                children && children.forEach(c => {
-                  persistShape(e.getShape(c), newWhiteboardId)
-                });
-              });
-              return;
-            }
+  const onPatch = (s, reason) => {
+    if (reason && isPresenter && (reason.includes("zoomed") || reason.includes("panned"))) {
+      if (cameraFitSlide.zoom === 0) {
+        //can happen when the slide finish uploading
+        const cameraFitSlide = calculateCameraFitSlide();
+        tldrawAPI?.setCamera(cameraFitSlide.point, cameraFitSlide.zoom);
+        setIsZoomed(false);
+        setCameraFitSlide(cameraFitSlide);
+        return;
+      }
+      const camera = tldrawAPI.getPageState().camera;

```

Comment:
is there a case where `tldrawAPI.getPageState()` is null or undefined?

Reply:
updated the operator in the event it's `null` or `undefined`